### PR TITLE
Simplify implementations of IApiTypeContainer.getPackageNames()

### PR DESCRIPTION
**Pull request by HannesWell**
and ensure the names are always unique and sorted.

### CODE REVIEW

File: apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/ArchiveApiTypeContainer.java [hunk 1]
```diff
@@ -238,7 +238,7 @@ public String[] getPackageNames() throws CoreException {
 		init();
 		synchronized (this) {
 			if (fPackageNames == null) {
-				fPackageNames = fPackages.keySet().toArray(String[]::new);
```
Comment (HannesWell) [inline apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/ArchiveApiTypeContainer.java:241]: @merks in b973ca8f012ff6de044b90485f38685bc3fef8b6 you changed this from sorted to not sorted. Was this intentional?

Comment (merks) [reply apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/ArchiveApiTypeContainer.java:241]: It's initialized by `fPackages = new TreeMap<>();`.

Comment (HannesWell) [reply apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/ArchiveApiTypeContainer.java]: May bad. Thanks for the clarification! Checked the other implementations as well, but that's the only one where sorting is not necessary.

**Comment by HannesWell** [comment]
Jenkins succeeds and on GH there are only the usual 349 failures in the API-tools tests, thus ready for submission.

**Pull request merged**